hostplumber/pkg/utils/sriov/sriov.go

```diff
@@ -0,0 +1,309 @@
+package sriov
+
+import (
+	"fmt"
+	"hostplumber/pkg/consts"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+func GetTotalVfs(devicePath string) int {
+	totalVfsFile := filepath.Join(devicePath, "sriov_totalvfs")
+	fd, err := os.Open(totalVfsFile)
+	defer fd.Close()
+	if err != nil {
+		return -1
+	}
+
+	var totalVfs int
+	_, err = fmt.Fscanf(fd, "%d\n", &totalVfs)
+	if err != nil {
+		return -1
+	}
+	return totalVfs
+}
+
+func GetNumVfs(devicePath string) int {
+	numVfsFile := filepath.Join(devicePath, "sriov_numvfs")
+	fd, err := os.Open(numVfsFile)
+	defer fd.Close()
+	if err != nil {
+		return -1
+	}
+
+	var numVfs int
+	_, err = fmt.Fscanf(fd, "%d\n", &numVfs)
+	if err != nil {
+		return -1
+	}
+	return numVfs
+}
+
+func GetVfPciAddrById(pfDevicePath string, vfId int) string {
+	virtFn := fmt.Sprintf("virtfn%d", vfId)
+	vfDeviceLink := filepath.Join(pfDevicePath, virtFn)
+	vfDevicePath, _ := filepath.EvalSymlinks(vfDeviceLink)
+	pciAddr := filepath.Base(vfDevicePath)
+	return pciAddr
+}
+
+func GetVfDriverByPci(vfPciAddr string) string {
+	driverLink := filepath.Join(consts.SysPciDevices, vfPciAddr, "driver")
+	driverPath, _ := filepath.EvalSymlinks(driverLink)
+	driverName := filepath.Base(driverPath)
+	return driverName
+}
+
+func GetPfDeviceForVf(devicePath string) (string, bool) {
+	physfnFile := filepath.Join(devicePath, "physfn")
+	if _, err := os.Lstat(physfnFile); err != nil {
+		return "", false
+	}
+	pfDevicePath, _ := filepath.EvalSymlinks(physfnFile)
+	return pfDevicePath, true
+}
+
+func VerifyPfExists(pfName string) bool {
+	_, err := os.Lstat(filepath.Join(consts.SysClassNet, pfName))
+	return err == nil
+}
+
+func GetTotalVfsForPf(pfName string) (int, error) {
+	totalVfsFile := filepath.Join(consts.SysClassNet, pfName, "device", "sriov_totalvfs")
+	fd, err := os.Open(totalVfsFile)
```

Comment:
Missing a close on the fd. (I think golint or govet should catch these if you want to check everything for this issue)